pysat/tests/classes/cls_instrument_iteration.py

```diff
@@ -0,0 +1,789 @@
+                    assert trange[1] < check, \
+                        "Stop time higher than expected"
+                elif i == (len(out['observed_times']) - 1):
+                    # Check that last load is at start of bounds.
+                    assert trange[0] == out['starts'][b_range], \
+                        "Loaded start time is not correct"
+                    assert trange[1] > out['starts'][b_range], \
+                        "End time lower than expected"
+                    assert trange[1] < out['starts'][b_range] + out['width'], \
+                        "End time higher than expected"
+
+        return
+
+    @pytest.mark.parametrize("operator", [('next'), ('prev')])
+    def test_file_load_empty_iteration(self, operator):
+        """Ensure empty iteration list is fine via day iteration."""
+
+        self.testInst.bounds = (None, None, '10000D',
+                                dt.timedelta(days=10000))
+        with pytest.raises(StopIteration) as err:
+            getattr(self.testInst, operator)()
+        estr = 'File list is empty. '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("first,second", [('next', 'prev'),
+                                              ('prev', 'next')])
+    def test_passing_bounds_with_iteration(self, first, second):
+        """Test if passing bounds raises StopIteration."""
+
+        # Load first data
+        getattr(self.testInst, first)()
+        with pytest.raises(StopIteration) as err:
+            # Iterate to a day outside the bounds.
+            getattr(self.testInst, second)()
+        assert str(err).find("Outside the set date boundaries") >= 0
+        return
+
+    def test_set_bounds_with_frequency(self):
+        """Test setting bounds with non-default step."""
+
+        start = self.ref_time
+        stop = self.ref_time + dt.timedelta(days=14)
+        self.testInst.bounds = (start, stop, 'M')
+        assert np.all(self.testInst._iter_list
+                      == pds.date_range(start, stop, freq='M').tolist())
+        return
+
+    def test_iterate_bounds_with_frequency(self):
+        """Test iterating bounds with non-default step."""
+
+        start = self.ref_time
+        stop = self.ref_time + dt.timedelta(days=15)
+        self.testInst.bounds = (start, stop, '2D')
+        self.eval_iter_list(start, stop, dates=True, freq=2)
+        return
+
+    def test_set_bounds_with_frequency_and_width(self):
+        """Set date bounds with step/width > 1."""
+
+        start = self.ref_time
+        stop = self.ref_time + pds.DateOffset(months=11, days=25)
+        stop = stop.to_pydatetime()
+        self.testInst.bounds = (start, stop, '10D', dt.timedelta(days=10))
+        assert np.all(self.testInst._iter_list
+                      == pds.date_range(start, stop, freq='10D').tolist())
+        return
+
+    @pytest.mark.parametrize("values", [(dt.datetime(2009, 1, 1),
```

Comment:
Ok, I have been thinking about this and think I have a more robust way to specify the start/stop times in `values`.  

1) In `values` specify dt.timedelta and keep the last two items unchanged
2) Inside(?) of `support_iter_evaluations` set start and stop time thusly:

```
start = self.testInst.inst_module._test_dates[''][''] 
stop = start + values[1]
```

Reply:
I'd prefer to bump this downstream since another rewrite of this section is coming in #902.  I've updated #863 to include this area since this is a known issue elsewhere.

Reply:
That's fine.  As long as the info is captured and an update is planned.

Reply:
Added a link to the above comment in #902 so this isn't lost.